packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/List/PreviewItem.tsx

```diff
@@ -0,0 +1,221 @@
+import {FieldPresence} from '../../../../../presence'
+import {useChildValidation} from '../../../../studio/contexts/Validation'
+import {ChangeIndicator} from '../../../../../changeIndicators'
+import {RowLayout} from '../../layouts/RowLayout'
+import {createProtoArrayValue} from '../createProtoArrayValue'
+import {InsertMenu} from '../InsertMenu'
+import {EditPortal} from '../../../../components/EditPortal'
+
+type PreviewItemProps<Item extends ObjectItem> = Omit<ObjectItemProps<Item>, 'renderDefault'>
+
+function getTone({
+  readOnly,
+  hasErrors,
+  hasWarnings,
+}: {
+  readOnly: boolean | undefined
+  hasErrors: boolean
+  hasWarnings: boolean
+}): CardTone {
+  if (readOnly) {
+    return 'transparent'
+  }
+  if (hasErrors) {
+    return 'critical'
+  }
+  return hasWarnings ? 'caution' : 'default'
+}
+const MENU_POPOVER_PROPS = {portal: true, tone: 'default'} as const
+
+export function PreviewItem<Item extends ObjectItem = ObjectItem>(props: PreviewItemProps<Item>) {
+  const {
+    schemaType,
+    parentSchemaType,
+    path,
+    readOnly,
+    onRemove,
+    value,
+    open,
+    onInsert,
+    onFocus,
+    onOpen,
+    onClose,
+    changed,
+    focused,
+    children,
+    inputProps: {renderPreview},
+  } = props
+
+  const sortable = parentSchemaType.options?.sortable !== false
+  const insertableTypes = parentSchemaType.of
+
+  const previewCardRef = useRef<HTMLDivElement | null>(null)
+
+  // this is here to make sure the item is visible if it's being edited behind a modal
+  useScrollIntoViewOnFocusWithin(previewCardRef, open)
+
+  useDidUpdate(focused, (hadFocus, hasFocus) => {
+    if (!hadFocus && hasFocus && previewCardRef.current) {
+      // Note: if editing an inline item, focus is handled by the item input itself and no ref is being set
+      previewCardRef.current?.focus()
+    }
+  })
+
+  const resolvingInitialValue = (value as any)._resolvingInitialValue
+
+  const handleDuplicate = useCallback(() => {
+    onInsert({
+      items: [{...value, _key: randomKey()}],
+      position: 'after',
+    })
+  }, [onInsert, value])
+
+  const handleInsert = useCallback(
+    (pos: 'before' | 'after', insertType: SchemaType) => {
+      onInsert({
+        items: [createProtoArrayValue(insertType)],
+        position: pos,
+      })
+    },
+    [onInsert]
+  )
+
+  const childPresence = useChildPresence(path, true)
+  const presence = useMemo(() => {
+    return childPresence.length === 0 ? null : (
+      <FieldPresence presence={childPresence} maxAvatars={1} />
+    )
+  }, [childPresence])
+
+  const childValidation = useChildValidation(path, true)
+  const validation = useMemo(() => {
+    return childValidation.length === 0 ? null : (
+      <Box marginLeft={1} paddingX={1} paddingY={3}>
+        <FormFieldValidationStatus validation={childValidation} __unstable_showSummary />
+      </Box>
+    )
+  }, [childValidation])
+
+  const hasErrors = childValidation.some((v) => v.level === 'error')
```

Comment:
These two are only used for getTone, so you _could_ consider making getTone take childeValidation instead, and remove 2 variables.